src/globalvars.cpp

```diff
@@ -90,4 +139,77 @@ void update_global_vars(const struct bpf_object *bpf_object,
   }
 }
 
+static uint32_t get_custom_rw_section_desired_size(
+    std::string_view section_name,
+    BPFtrace &bpftrace)
+{
+  if (section_name == FMT_STRINGS_BUFFER_SECTION_NAME) {
+    return bpftrace.ncpus_ * bpftrace.resources.max_fmtstring_args_size;
+  }
+  LOG(BUG) << "Unknown section " << section_name;
+  // unreachable
+  return 0;
+}
+
+static void update_global_vars_custom_rw_section(
+    const struct bpf_object *bpf_object,
+    std::string_view section_name,
+    struct bpf_map *global_vars_map,
+    const std::unordered_set<std::string> &needed_global_vars,
+    BPFtrace &bpftrace)
+{
+  auto desired_size = get_custom_rw_section_desired_size(section_name,
+                                                         bpftrace);
+  auto err = bpf_map__set_value_size(global_vars_map, desired_size);
+  if (err != 0) {
+    LOG(BUG) << "Failed to set size to " << desired_size << " for section "
+             << section_name;
+  }
```

Comment:
Not necessarily a bug, right? Maybe libbpf or kernel thinks it's too big. Probably want to throw `FatalUserException`: https://github.com/bpftrace/bpftrace/blob/master/docs/coding_guidelines.md#error-handling-and-exceptions